os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -391,9 +405,10 @@ static int amebasmart_i2s_tx(struct amebasmart_i2s_s *priv, struct amebasmart_bu
 		}
 		apb->curbyte += tx_size; /* No padding, ptx_buf is big enough to fill the whole tx_size */
 
+		i2s_enable(&priv->i2s_object);
 		i2s_send_page(&priv->i2s_object, (uint32_t *)ptx_buf);
 	}
-	return OK;
+	return ret;
```

Comment:
It always returns OK. It it ok to return OK even `bfcontainer` or `bfcontainer->apb` is NULL?

Reply:
Added -1 and i2serr message